### PR TITLE
Fix tool attributes mining levels

### PR DESCRIPTION
**Pull request by snakefangox**
Currently mining levels are ignored in a few situations, namely when using a Dynamic tool and when using vanilla tools on modded blocks. Modded tools also behave differently on blocks they can't mine vs vanilla tools.

This PR fixes those bugs, the associated issues are #853 and #841

### CODE REVIEW

**Review by hYdos — APPROVED** [review]

**Comment by Player3324** [comment]
Such an incompatible change is not desirable, please try to retain the old method.  This also needs tests that expose and verify the missing cases presumably fixed by this PR.

**Comment by snakefangox** [comment]
My bad, I've re-added the method I removed and changed the version bump to match. I've also added tests to the test mod to catch the bugs involved.

**Review by Player3324 — APPROVED** [review]
Looks good to me, thanks!

**Review by liach — APPROVED** [review]

File: fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java [hunk 1]
```diff
@@ -18,63 +18,122 @@
...
-import net.fabricmc.fabric.api.event.server.ServerTickCallback;
 
 public class ToolAttributeTest implements ModInitializer {
-	private boolean hasValidatedTags = false;
+	private static final float DEFAULT_BREAK_SPEED = 1.0F;
+	private static final float TOOL_BREAK_SPEED = 10.0F;
+
+	private boolean hasValidated = false;
+
+	Block gravel_block;
```
Comment (shedaniel) [inline fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java]: minor bikeshed but probably want to use the normal names? such as `gravelBlock`

File: fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java [hunk 2]
```diff
@@ -18,63 +18,122 @@
...
 		}
 
-		hasValidatedTags = true;
+		hasValidated = true;
 
 		if (FabricToolTags.PICKAXES.values().isEmpty()) {
 			throw new AssertionError("Failed to load tool tags");
 		}
+
+		//Test vanilla tools don't bypass fabric mining levels
```
Comment (shedaniel) [inline fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java:98]: possibly want to test for vanilla tools for vanilla blocks

**Review by shedaniel — APPROVED** [review]
looks good other some minor nitpicks

File: fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java (hunk 1 above)
Comment (snakefangox) [reply fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java]: Yeah, can't remember why I did that so I've changed to match the standard.

File: fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java (hunk 2 above)
Comment (snakefangox) [reply fabric-tool-attribute-api-v1/src/testmod/java/net/fabricmc/fabric/test/tool/attribute/ToolAttributeTest.java:98]: Good idea, I've added all vanilla test cases.

**Review by shedaniel — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Review by haykam821 — APPROVED** [review]

**Review by vini2003 — APPROVED** [review]

**Pull request merged**